### ISSUE TITLE
publish additional JARs for 2.13.0-M5?

### ISSUE THREAD

**Issue by SethTisue** [lloydmeta/enumeratum#216]
followup to #199 

a lot more of your dependencies are now available, so it should be possible to publish e.g. enumeratum-circe, which we wanted over at https://github.com/pauljamescleary/scala-pet-store/pull/142

**Comment by lloydmeta** [comment]
I just released the circe integration; with the new Play 2.7 integration that just got merged, I _think_ that's all for now?

Will keep this ticket open to track others.

**Comment by Milyardo** [comment]
publishing for the release candidate now would be a great idea

**Comment by lloydmeta** [comment]
> publishing for the release candidate now would be a great idea

Ah ok, looks like RC1 is a thing now https://github.com/scala/scala/milestones?state=closed

Unfortunately, we're a bit blocked at the moment on https://github.com/scalatest/scalatest/issues/1567 for the main core/macro projects. Very likely the same situation w/ the integration libs, but I'm more comfortable having breakage there in a separate branch for now as those projects start publishing artefacts for `2.13.0-RC1`.

Will keep an eye out though.

**Comment by SethTisue** [comment]
> we're a bit blocked at the moment on scalatest/scalatest#1567

ScalaTest 3.0.8-RC2 has been published for Scala 2.13.0-RC1, is that sufficient for you to go forward with?

**Comment by lloydmeta** [comment]
> > we're a bit blocked at the moment on [scalatest/scalatest#1567](https://github.com/scalatest/scalatest/issues/1567)
> 
> as per https://github.com/scala/make-release-notes/blob/2.13.x/projects-2.13.md, ScalaTest 3.0.8-RC2 has been published for Scala 2.13.0-RC1, is that sufficient for you to go forward with?

Yeah, I've now got a branch that tries to get us there https://github.com/lloydmeta/enumeratum/tree/build/2.13-rc1

I have a few more deprecation warnings and such to work through though.

**Comment by guilgaly** [comment]
Hi, is there any news on this ? It looks like 2.13.0 final [might be here very soon](https://contributors.scala-lang.org/t/scala-2-13-0-rc2-release/3323/10).

**Comment by lloydmeta** [comment]
> Hi, is there any news on this ? It looks like 2.13.0 final [might be here very soon](https://contributors.scala-lang.org/t/scala-2-13-0-rc2-release/3323/10).

Nope, you're welcome to contribute to pushing https://github.com/lloydmeta/enumeratum/tree/build/2.13-rc1 over the finish line though (or start a new branch :) )

Been super busy with work for the past few weeks 😢

**Comment by guilgaly** [comment]
I decided to try to see what I could do, I've made some progress on https://github.com/guilgaly/enumeratum/tree/build/2.13-rc2 (starting from your branch). 🎉 

- Of the projects that were already in the `scala_2_13` project, only `enumeratum-play` seems to be blocked by a missing dependency (`scalatestplus-play`).
- I still have the `method enclosingClass in trait Enclosures is deprecated (since 2.11.0)` warning, not sure how to get rid of it like on 2.11/2.12...
- For some reason, on Scala 2.13-RC2, the wartremover rule `StringPlusAny` also triggers inside string interpolations (see wartremover/wartremover#447), so I had to add some explicit `toString`s there.

**Comment by lloydmeta** [comment]
That's awesome ! 

I think the `method enclosingClass in trait Enclosures is deprecated (since 2.11.0)` is going to always be there; AFAIK, there's no alternative offered that truly replaces it, and we need to use it for ensuring as-declared-ordering discovery of subclasses. I think for 2.13.x, we can just disable deprecation.

Whenever you're done, please open a PR :) If it doesn't pass the tests due to some kind of missing dependency on an integration module, we can maybe merge it into the existing https://github.com/lloydmeta/enumeratum/tree/build/2.13-rc1  branch ( I know you did rc2, but eh :D), and keep it there until all the integration modules are ready?

**Comment by lloydmeta** [comment]
FYI, RC2 builds have been published, thanks to @guilgaly :)

**Comment by lloydmeta** [comment]
Closing this now that 2.13 has been released, `-M5`, and enumeratum artefacts have been published for 2.13 is no longer relevant.
